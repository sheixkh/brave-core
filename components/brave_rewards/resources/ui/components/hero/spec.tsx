/* global jest, expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import Hero from './index'
import { TestThemeProvider } from 'brave-ui/theme'

describe('Hero tests', () => {
  const baseComponent = (props?: object) => <TestThemeProvider><Hero id='test-hero' {...props} /></TestThemeProvider>

  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const component = baseComponent()
      const tree = create(component).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the component', () => {
      const wrapper = shallow(baseComponent())
      const assertion = wrapper.find('#test-hero').length
      expect(assertion).toBe(1)
    })
  })
})
