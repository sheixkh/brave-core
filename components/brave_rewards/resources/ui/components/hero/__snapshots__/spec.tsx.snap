// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero tests basic tests matches the snapshot 1`] = `
<div
  className="sc-bdVaJa izGEVs"
  id="test-hero"
/>
`;
